RHIDP-5367 - Missing and incorrect information in "breaking changes" (#827)
redhat-developer/red-hat-developers-documentation-rhdh: modules/release-notes/ref-release-notes-breaking-changes.adoc

[id="feature-rhidp-4853"]
== Plugins with updated scope

> With this update, the following plugins that were previously under the `@janus-idp` scope have moved to `@backstage-community`:

Reviewer: ```suggestion

If you use any of the plugins listed below, you must revise your configuration to use the latest plugins from the new scope in order to upgrade from {prod-short} 1.3 to 1.4.

With this update, the following plugins that were previously under the `@janus-idp` scope have moved to `@backstage-community`:
```

[cols=2,%header]
|===

> | *Plugin Name*

Reviewer: ```suggestion
| *Old Plugin Name*
```

Author: @nickboldt For consistency (with RHDH 1.3 RNs), how about we use:
**RHDH 1.3 Plugin Name** and **RHDH 1.4 Plugin Name** in place of **Old Plugin Name** and **New Plugin Name**

Reviewer: sure, works for me

> | *Plugin Name*

Reviewer: ```suggestion
| *New Plugin Name*
```

|`@janus-idp/backstage-plugin-nexus-repository-manager`|`@backstage-community/plugin-nexus-repository-manager`
|`@janus-idp/backstage-plugin-ocm`|`@backstage-community/plugin-ocm`
|`@janus-idp/backstage-plugin-ocm-backend`|`@backstage-community/plugin-ocm-backend`
|`@janus-idp/backstage-plugin-quay`|`@backstage-community/plugin-quay`
|`@janus-idp/backstage-plugin-rbac`|`@backstage-community/plugin-rbac`
|`@janus-idp/backstage-plugin-tekton`|`@backstage-community/plugin-tekton`
|`@janus-idp/backstage-plugin-topology`|`@backstage-community/plugin-topology`
|`@janus-idp/backstage-scaffolder-backend-module-quay`|`@backstage-community/plugin-scaffolder-backend-module-quay`
|`@janus-idp/backstage-scaffolder-backend-module-regex`|`@backstage-community/plugin-scaffolder-backend-module-regex`
|`@janus-idp/backstage-scaffolder-backend-module-servicenow`|`@backstage-community/plugin-scaffolder-backend-module-servicenow`
|`@janus-idp/backstage-scaffolder-backend-module-sonarqube`|`@backstage-community/plugin-scaffolder-backend-module-sonarqube`
|===

The following plugins that were previously under the `@backstage` scope have moved to `@backstage-community`:
[cols=2,%header]
|===

> | *Plugin Name*

Reviewer: ```suggestion
| *Old Plugin Name*
```

> | *Plugin Name*

Reviewer: ```suggestion
| *New Plugin Name*
```

|`@backstage/plugin-azure-devops`|`@backstage-community/plugin-azure-devops`
|`@backstage/plugin-azure-devops-backend`|`@backstage-community/plugin-azure-devops-backend`
|`@backstage/plugin-dynatrace`|`@backstage-community/plugin-dynatrace`
|`@backstage/plugin-github-actions`|`@backstage-community/plugin-github-actions`
|`@backstage/plugin-github-issues`|`@backstage-community/plugin-github-issues`
|`@backstage/plugin-jenkins`|`@backstage-community/plugin-jenkins`
|`@backstage/plugin-jenkins-backend`|`@backstage-community/plugin-jenkins-backend`
|`@backstage/plugin-lighthouse`|`@backstage-community/plugin-lighthouse`
|`@backstage/plugin-sonarqube`|`@backstage-community/plugin-sonarqube`
|`@backstage/plugin-sonarqube-backend`|`@backstage-community/plugin-sonarqube-backend`
|`@backstage/plugin-tech-radar`|`@backstage-community/plugin-tech-radar`
|===

Two plugins previously under the @janus-idp scope have moved to @red-hat-developer-hub:

[cols=2,%header]
|===

> | *Plugin Name*

Reviewer: ```suggestion
| *Old Plugin Name*
```

> | *Plugin Name*

Reviewer: ```suggestion
| *New Plugin Name*
```

| `@red-hat-developer-hub/backstage-plugin-bulk-import-backend`
|===

As the scope of the previous plugins has been updated, the dynamic plugin configuration has also changed.

[cols=2,%header]
|===
|*RHDH 1.3 Configuration*|*RHDH 1.4 Configuration*
|link:https://github.com/janus-idp/backstage-showcase/blob/release-1.3/dynamic-plugins.default.yaml[dynamic-plugins.default.yaml]|link:https://github.com/janus-idp/backstage-showcase/blob/release-1.4/dynamic-plugins.default.yaml[dynamic-plugins.default.yaml]
|===

.Procedure

> * If your workload requires plugins with an updated scope, revise your configuration to use the latest plugins from the new scope.

Reviewer: ```suggestion
* If you use any of the plugins listed above, revise your configuration to use the latest plugins from the new scope.

In addition to the tables above, you can also see which dynamic plugins have changed between {prod-short} 1.3 and 1. by comparing these two files:

* https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/release-1.3/modules/dynamic-plugins/rhdh-supported-plugins.csv
* https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/release-1.4/modules/dynamic-plugins/rhdh-supported-plugins.csv
```

| `@red-hat-developer-hub/backstage-plugin-bulk-import-backend`
|===

With the update to the plugin scope, the dynamic plugin configuration has also been modified.

[cols=2,%header]
|===
|*RHDH 1.3 Configuration*|*RHDH 1.4 Configuration*
|link:https://github.com/janus-idp/backstage-showcase/blob/release-1.3/dynamic-plugins.default.yaml[dynamic-plugins.default.yaml]|link:https://github.com/janus-idp/backstage-showcase/blob/release-1.4/dynamic-plugins.default.yaml[dynamic-plugins.default.yaml]
|===

.Procedure
* To upgrade from {product-very-short} 1.3 to {product-very-short} 1.4, you must update your configuration to use the latest versions of the plugins listed previously from the new scope.

[NOTE]
====

> Along with the tables provided previously, you can compare the following files to see which dynamic plugins have changed between {product-very-short} 1.3 and {product-very-short} 1.4:

Reviewer 2: The added URLs don't seem very user-friendly. Would it be okay to revise the sentence as follows, adding text to the links and removing the bullet points?
```suggestion
In addition to the previously provided tables, you can compare the link:https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/release-1.4/modules/dynamic-plugins/rhdh-supported-plugins.csv[RHDH 1.4 CSV file] with the link:https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/release-1.3/modules/dynamic-plugins/rhdh-supported-plugins.csv[RHDH 1.3 CSV file] to identify the changes in dynamic plugins.
```

> In addition to the tables above, you can also review which dynamic plugins have changed between {product-very-short} 1.3 and {product-very-short} 1.4 by comparing these two files:

Reviewer 2: As per IBMSG, instead of using above we can use previous or preceding:
![image](https://github.com/user-attachments/assets/6ecdcebf-1e11-441f-8e5e-c11b276df900)

```suggestion
Along with the tables provided previously, you can compare the following files to see which dynamic plugins have changed between {product-very-short} 1.3 and {product-very-short} 1.4:
```

> * https://github.com/redhat-developer/red-hat-developers-documentation-rhdh/blob/release-1.3/modules/dynamic-plugins/rhdh-supported-plugins.csv

Reviewer 2: Shouldn't we add a link text to these CSV files?

Author: The URLs provided are clickable.
